pkgs/esp-idf/default.nix

```diff
@@ -38,6 +38,7 @@ let
     rev = rev;
     sha256 = sha256;
     fetchSubmodules = true;
+    leaveDotGit = true;
```

Comment:
Be aware, that you also need to change the hash, if you decide to leave the `.git` directory in there.